nx/lib/nx.ex

```diff
@@ -3355,18 +3355,36 @@ defmodule Nx do
           [11]
         ]
       >
+
+  Negative indices are also accepted, in the same fashion as `Enum.split/2`.
+
+      iex> t = Nx.tensor([1, 2, 3, 4])
+      iex> {left, right} = Nx.split(t, -1)
+      iex> left
+      #Nx.Tensor<
+        s64[3]
+        [1, 2, 3]
+      >
+      iex> right
+      #Nx.Tensor<
+        s64[1]
+        [4]
+      >
   """
   @doc type: :indexed
   def split(tensor, split, opts \\ [])
 
-  def split(tensor, split, opts) do
+  def split(tensor, split, opts) when is_number(split) do
```

Comment:
If you add this guard, then you can remove `true ->` from the `cond`, but I think the `cond` is clearer. So I would not change this here.